### PR TITLE
Avoid scroll in Add Strings to screenshot

### PR DESCRIPTION
**Pull request by meel-hd**
## Proposed changes
**Avoid scroll in Add Strings to screenshot**

- Added a height limit to the first table showing strings related to the screenshot. to prevent scroll and hiding the second table.
- Fixed buttons overallaping in the actions of the tables on mobile devices

Fixes: #7085

## Checklist

- [x] Lint and unit tests pass locally with my changes.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added documentation to describe my feature.
- [x] I have squashed my commits into logic units.
- [x] I have described the changes in the commit messages.

## Summary by CodeRabbit

- **New Features**
	- Introduced a new CSS file for improved styling of the sources listing component, enhancing usability and aesthetics, especially on mobile devices.
	- Added a flexible `extra_styles` block in the HTML templates to allow for custom CSS in derived templates, improving styling options for the screenshot detail page.

- **Style**
	- Enhanced layout and spacing for the footer and buttons in the sources listing component.
	- Updated the search input element for better styling consistency.

### CODE REVIEW

File: weblate/static/styles/screenshots/screenshot_detail.css [hunk 1]
```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright © Michal Čihař <michal@weblate.org>
+ *
+ * SPDX-License-Identifier: GPL-3.0-or-later
+ */
+
+#sources-listing {
```
Comment (nijel) [inline weblate/static/styles/screenshots/screenshot_detail.css:7]: I think it would look nicer if only the inner table was scrollable, not the full panel.

**Review by nijel — COMMENTED** [review]
Looks better now. Can the margin below the table be removed? I think it was removed by bootstrap when the table was directly in the panel.

**Comment by meel-hd** [comment]
> Can the margin below the table be removed?

Yes. But there will be no separation between it and the below element.
Is that what is desired? To hint the two tables are related. If otherwise I don't think it will look good.

**Comment by nijel** [comment]
I mean this:

image

The space is there only if the table is scrolled up, so it really doesn't serve as a separator.

Before scrolling down, it is correct:

image

File: weblate/static/styles/screenshots/screenshot_detail.css [hunk 2]
```diff
@@ -4,6 +4,10 @@
  * SPDX-License-Identifier: GPL-3.0-or-later
  */
 
+ #sources-listing {
+  margin-bottom: 0px;
+ }
```
Comment (nijel) [inline weblate/static/styles/screenshots/screenshot_detail.css]: This is not what I meant, see the screenshots I've just posted.

Comment (meel-hd) [reply weblate/static/styles/screenshots/screenshot_detail.css]: Uh I see. Fixing it.

**Comment by nijel** [comment]
Merged, thanks for your contribution!

**Pull request merged**